crypto/chacha/chacha.c

```diff
@@ -41,9 +41,21 @@ static const uint8_t sigma[16] = { 'e', 'x', 'p', 'a', 'n', 'd', ' ', '3',
 void CRYPTO_hchacha20(uint8_t out[32], const uint8_t key[32],
                       const uint8_t nonce[16]) {
   uint32_t x[16];
+#ifdef OPENSSL_BIG_ENDIAN
+  for(size_t i = 0; i < 4; i++) {
+    x[i] = CRYPTO_load_u32_le(sigma + i * sizeof(uint32_t));
```

Comment:
if performance is of any concern we could define two versions of `sigma` (be and le) and do only memcopy.